docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,663 @@
+and the section later on the [Override Layer](#override-layer).
+
+Meta-layers are detailed more in the
+[Meta-Layers](LoaderLayerInterface.md#meta-layers) section of the
+[LoaderLayerInterface.md](LoaderLayerInterface.md) file in this folder.
+
+
+### Implicit vs Explicit Layers
+
+![Different Types of Layers](./images/loader_layer_order.png)
+
+Explicit layers are layers which are enabled by an application (e.g. with the
+vkCreateInstance function as mentioned previously), or by an environment
+variable.
+
+Implicit layers are those which are enabled by their existence.
+For example, certain application environments (e.g. Steam or an automotive
+infotainment system) may have layers which they always want enabled for all
+applications that they start.
+Other implicit layers may be for all applications started on a given system
+(e.g. layers that overlay frames-per-second).
+Implicit layers are enabled automatically, whereas explicit layers must be
+enabled explicitly.
+
+Implicit layers have an additional requirement over explicit layers in that
+they require being able to be disabled by an environmental variable.
+This is due to the fact that they are not visible to the application and could
+cause issues.
+A good principle to keep in mind would be to define both an enable and disable
+environment variable so the users can deterministically enable the
+functionality.
+On Desktop platforms (Windows, Linux, and macOS), these enable/disable settings
+are defined in the layer's JSON file.
+
+Discovery of system-installed implicit and explicit layers is described later
+in the [Layer discovery](LoaderLayerInterface#layer-discovery)
+section in the
+[LoaderLayerInterface.md document](LoaderLayerInterface.md) document.
+For now, simply know that what distinguishes a layer as implicit or explicit
+is dependent on the Operating system, as shown in the table below.
+
+<table style="width:100%">
+  <tr>
+    <th>Operating System</th>
+    <th>Implicit Layer Identification</th>
+  </tr>
+  <tr>
+    <td>Windows</td>
+    <td>Implicit layers are located in a different Windows registry location
+        than explicit layers.</td>
+  </tr>
+  <tr>
+    <td>Linux</td>
+    <td>Implicit layers are located in a different directory location than
+        explicit layers.</td>
+  </tr>
+  <tr>
+    <td>Android</td>
+    <td>There is **No Support For Implicit Layers** on Android.</td>
+  </tr>
+  <tr>
+    <td>macOS</td>
+    <td>Implicit layers are located in a different directory location than
+        explicit layers.</td>
+  </tr>
+</table>
+
+
+#### Override Layer
+
+The "Override Layer" is a special implicit meta-layer created by the 
+[VkConfig](https://github.com/LunarG/VulkanTools/blob/master/vkconfig/README.md)
+tool and available by default when the tool is running.
+Once VkConfig exits, the override layer is removed, and the system should
+return to standard Vulkan behavior.
```

Comment:
We should remove the loader code responsible for the 'expiration' because that was a silly idea that doesn't live on in the current vkconfig application.

Reply:
Kind of orthogonal to this.  I don't think I have any mention of that expiring behavior in this doc, do I?

Reply:
Yeah, definitely is orthogonal. Very much a low priority task, since its not affecting anyone (that we know of)